gossip/src/crds_value.rs

```diff
@@ -485,6 +498,69 @@ impl Sanitize for NodeInstance {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Default, PartialEq, Eq, AbiExample)]
+pub struct RestartLastVotedForkSlots {
+    pub slots: EpochSlots,
+    pub last_voted_slot: Slot,
+    pub last_voted_hash: Hash,
+}
+
+impl Sanitize for RestartLastVotedForkSlots {
+    fn sanitize(&self) -> std::result::Result<(), SanitizeError> {
+        self.slots.sanitize()?;
+        self.last_voted_hash.sanitize()
+    }
+}
+impl fmt::Debug for RestartLastVotedForkSlots {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "RestartLastVotedForkSlots {{ slots: {:?} last_voted: {}({}) }}",
+            self.slots, self.last_voted_slot, self.last_voted_hash
+        )
+    }
+}
+
+impl RestartLastVotedForkSlots {
+    pub fn new(from: Pubkey, now: u64, last_voted_slot: Slot, last_voted_hash: Hash) -> Self {
+        Self {
+            slots: EpochSlots {
+                from,
+                slots: vec![],
+                wallclock: now,
+            },
+            last_voted_slot,
+            last_voted_hash,
+        }
+    }
+
+    pub fn new_from_fields(
+        slots: EpochSlots,
+        last_voted_slot: Slot,
+        last_voted_hash: Hash,
+    ) -> Self {
+        Self {
+            slots,
+            last_voted_slot,
+            last_voted_hash,
+        }
+    }
```

Comment:
If all fields are `pub` this is redundant.

Reply:
Removed.